src/foundry/client-esm/audio/sound.d.mts

```diff
@@ -0,0 +1,401 @@
+import type { AnyObject, InexactPartial, ValueOf } from "../../../types/utils.d.mts";
+import type EventEmitterMixin from "../../common/utils/event-emitter.d.mts";
+
+declare namespace Sound {
+  interface PlaybackOptions extends _PlaybackOptions {}
+
+  /** @internal */
+  type _PlaybackOptions = InexactPartial<{
+    /**
+     * A delay in seconds by which to delay playback
+     * @defaultValue `0`
+     */
+    delay: number;
+
+    /** A limited duration in seconds for which to play */
+    duration: number;
+
+    /**
+     * A duration in milliseconds over which to fade in playback
+     * @defaultValue `0`
+     */
+    fade: number;
+
+    /** Should sound playback loop? */
+    loop: boolean;
+
+    /** Seconds of the AudioBuffer when looped playback should start. Only works for AudioBufferSourceNode. */
+    loopStart: number;
+
+    /** Seconds of the Audio buffer when looped playback should restart. Only works for AudioBufferSourceNode. */
+    loopEn: number;
+
+    /** An offset in seconds at which to start playback */
+    offset: number;
+
+    /** A callback function attached to the source node */
+    onended: (...args: unknown[]) => void | null;
```

Comment:
so `(...args: unknown) => void | null` is a function that must _accept_ anything and returns either `void` or `null`. So you can't write `onended: null`. Based upon actual usage you meant `((...args: unknown[]) => void) | null`

Next, saying `(...args: unknown[]) => ...` actually forces you to give a function that can accept anything. Based upon actual usage though, `this.#playback.onended?.(this);`, this should be `(sound: Sound) => ...`